### ISSUE TITLE
Mautic Tracking Cookies set before Consent is given

### ISSUE THREAD

**Issue by dpw1460e** [AmauriC/tarteaucitron.js#453]
I'm using Version 1.4 of the tarteaucitron.js (Thanks for this Open Source Tool).

When I use the script on a Mautic Landing Page, it still allows the Mautic mtc_id and mtc_sid cookies to be set before visitor consent.

At that time the tarteaucitron cookie value is !mautic=wait. 

From my point of view it should be !mautic=false until the visitor has given consent to the tracking.

While testing I observed that I could deny the Mautic cookies and they will be deleted (even though they should have been created on page load). But when I allow the Mautic cookies later, they will not be created again (I guess because the page is not loaded again).

Right now I'm not sure if this issue could be solved through your script or if it has to be solved with Mautic. 

I just wanted make you aware of my findings. Maybe other Mautic Users have experienced this too.

Thanks a lot for your efforts.

Best,
Stephan

P.S. It would be a great advantage if links to the Privacy Policy or any other pages from the Cookie Consent Tool would be opened in a new window. Alternatively you might like to provide this ability.

**Comment by AmauriC** [comment]
Check the expiration date of the cookie, if this is in the past, this is just to be sure that cookies are deleted before the acceptation by the user. This is the point of the "wait" status, not sure if this is disabled but not yet activated.

**Comment by dpw1460e** [comment]
Here is what I did so far:
- created a new cookie called "priorconsent" with value "NoConsentYet"
- added a script to be run when above cookie is found and delete all Mautic Cookies (works)
- Now the visitor could use the Tarteaucirton,js options to give Consent to the Tracking.

The following is what I observed:
1) if Consent is denied the page is reloaded and the cookies are deleted.
2) If Consent is given no reload of the page and no tracking cookies are created.
3) if consent was denied at first and later given, no reload of the page and no cookies created.

Thus I'm now fiddling with reloading the page when the consent is given at first or later (thus the cookieconsent value is changed). I wonder why the script loads the page on consent denial, but not on consent given.

Though how do I force a page reload on changed Consent State?

Thanks for your help.

Cheers,
Stephan

**Comment by dpw1460e** [comment]
Here is the process of what I wanted:
1) delete all Cookies set by Mautic prior Visitor Consent
2) Set Cookies as per Visitor Consent (by using tarteaucitron.js)

The Solution:
1) Create a Cookie called "priorconsent" with value "NoConsentYet"
2) Check for the above cookie and state, if exits delete all Mautic Cookies and the cookie created under 1)
3) Set Cookies using tarteaucitron.js and refresh page on every Cookie change.

To 3) to work as I intended I needed to change:
tarteaucitron.js line 31 "reloadThePage": false, to "reloadThePage": true,
and needed to add window.location.reload(); to the onclick event for the "OK, accept all" button (lines 314 and 381).

That's it. Now it works as I like it to work.
Thanks for the awesome script. I really appreciate your efforts.

Cheers,
Stephan
